x-pack/plugin/mapper-constant-keyword/src/test/java/org/elasticsearch/xpack/constantkeyword/mapper/SearchIdleTests.java

```diff
@@ -0,0 +1,326 @@
+            .get();
+        assertEquals(RestStatus.OK, searchResponse.status());
+        assertEquals(idleIndexShardsCount, searchResponse.getSkippedShards());
+        assertEquals(0, searchResponse.getFailedShards());
+        Arrays.stream(searchResponse.getHits().getHits()).forEach(searchHit -> assertEquals("test2", searchHit.getIndex()));
+
+        // THEN
+        final IndicesStatsResponse idleIndexStatsAfter = client().admin().indices().prepareStats(idleIndex).get();
+        Arrays.stream(idleIndexStatsAfter.getShards()).forEach(shardStats -> assertTrue(shardStats.isSearchIdle()));
+
+        final IndicesStatsResponse activeIndexStatsAfter = client().admin().indices().prepareStats(activeIndex).get();
+        Arrays.stream(activeIndexStatsAfter.getShards()).forEach(shardStats -> assertFalse(shardStats.isSearchIdle()));
+
+        assertIdleShardsRefreshStats(idleIndexStatsBefore, idleIndexStatsAfter);
+        assertActiveShardsRefreshStats(activeIndexStatsBefore, activeIndexStatsAfter);
+    }
+
+    private CreateIndexResponse createIndex(final String idleIndex, int shardsCount, final String... mapping) {
+        return client().admin()
+            .indices()
+            .prepareCreate(idleIndex)
+            .setSettings(
+                Settings.builder()
+                    .put(IndexSettings.INDEX_SEARCH_IDLE_AFTER.getKey(), "500ms")
+                    .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, shardsCount)
+                    .put(IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.getKey(), -1)
+            )
+            .setMapping(mapping)
+            .get();
+    }
+
+    private SearchResponse search(final String index, final String field, final String value, int preFilterShardSize) {
+        return client().prepareSearch(index)
+            .setQuery(new MatchPhraseQueryBuilder(field, value))
+            .setPreFilterShardSize(preFilterShardSize)
+            .get();
+    }
+
+    private static void assertIdleShard(final IndicesStatsResponse statsResponse) {
+        Arrays.stream(statsResponse.getShards()).forEach(shardStats -> assertTrue(shardStats.isSearchIdle()));
+        Arrays.stream(statsResponse.getShards()).forEach(shardStats -> assertTrue(shardStats.getSearchIdleTime() >= 100));
+    }
+
+    private void assertActiveShardsRefreshStats(final IndicesStatsResponse before, final IndicesStatsResponse after) {
+        assertNotEquals(0, before.getShards().length);
+        assertNotEquals(0, after.getShards().length);
+        final Map<ShardRouting, ShardStats> afterStatsAsMap = after.asMap();
+        for (var beforeShard : before.getShards()) {
+            final RefreshStats beforeRefresh = beforeShard.getStats().refresh;
+            assertTrue(afterStatsAsMap.containsKey(beforeShard.getShardRouting()));
+            final RefreshStats afterRefresh = afterStatsAsMap.get(beforeShard.getShardRouting()).getStats().refresh;
+            assertTrue(afterRefresh.getTotal() >= beforeRefresh.getTotal());
```

Comment:
Here we should need just `>` (not equals) because a refresh should have happened in the meanwhile after executing the search query. Anyway, if I remove the equal the testing refresh stats on the active index fails because the values are the sane before and after. I am trying to figure out why.